dbt/include/sqlserver/macros/adapters.sql

```diff
@@ -2,6 +2,15 @@
   information_schema
 {%- endmacro %}
 
+{% macro sqlserver__get_columns_in_query(select_sql) %}
+    {% call statement('get_columns_in_query', fetch_result=True, auto_begin=False) -%}
+        select TOP 0 * from (
+            {{ select_sql }}
+        ) as __dbt_sbq
+    {% endcall %}
+    {{ return(load_result('get_columns_in_query').table.columns | map(attribute='name') | list) }}
+{% endmacro %}
```

Comment:
@mikael, this is one macro we had to overwrite. perhaps this is the error you were seeing?